src/libraries/System.Net.WebHeaderCollection/src/System/Net/WebHeaderCollection.cs

```diff
@@ -141,15 +129,7 @@ public override void Set(string name, string? value)
             }
 
             name = HttpValidationHelpers.CheckBadHeaderNameChars(name);
-            ThrowOnRestrictedHeader(name);
             value = HttpValidationHelpers.CheckBadHeaderValueChars(value);
-            if (_type == WebHeaderCollectionType.WebResponse)
-            {
-                if (value != null && value.Length > ushort.MaxValue)
-                {
-                    throw new ArgumentOutOfRangeException(nameof(value), value, SR.Format(CultureInfo.InvariantCulture, SR.net_headers_toolong, ushort.MaxValue));
```

Comment:
Are the `SR.net_headers_toolong` and `SR.net_headerrestrict` resource strings used anywhere else? Otherwise they can be removed

Reply:
Thanks, removed